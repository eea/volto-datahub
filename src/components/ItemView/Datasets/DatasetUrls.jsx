import { Icon, List } from 'semantic-ui-react';
import { SVGIcon } from '../utils.js';

import lockSVG from 'remixicon/icons/System/lock-line.svg';
import { isInternalURL } from '@eeacms/volto-datahub/utils';

export default function DatasetUrls({ dataset }) {
  return (dataset['WWW:URL'] || []).map((item, i) => {
    return (
      <List.Item key={i}>
        <List.Content>
          <div className="dataset-item">
            <Icon className="download" />
            <a
              href={item.url}
              className="item-link"
              target="_blank"
              rel="noreferrer"
            >
              <span>{item.name}</span>
            </a>
            {isInternalURL(item.url) && <SVGIcon name={lockSVG} size="20px" />}
          </div>
        </List.Content>
        {item.description && (
          <span className="item-description" title={item.description}>
            {item.description}
          </span>
        )}
      </List.Item>
    );
  });
}
