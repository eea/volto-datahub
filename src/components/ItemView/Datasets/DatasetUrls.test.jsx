import React from 'react';
import renderer from 'react-test-renderer';

import DatasetUrls from './DatasetUrls';

jest.mock('@eeacms/search', () => ({
  runRequest: jest.fn(),
}));

const dataset = {
  'WWW:URL': [
    {
      url: 'https://example.org/1',
      name: 'Direct download',
      description: 'Download link description',
    },
  ],
};

describe('DatasetUrl', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DatasetUrls dataset={dataset} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
