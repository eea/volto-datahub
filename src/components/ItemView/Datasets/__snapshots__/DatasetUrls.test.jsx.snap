// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetUrl renders correctly 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="content"
  >
    <div
      className="dataset-item"
    >
      <i
        aria-hidden="true"
        className="icon download"
        onClick={[Function]}
      />
      <a
        className="item-link"
        href="https://example.org/1"
        rel="noreferrer"
        target="_blank"
      >
        <span>
          Direct download
        </span>
      </a>
    </div>
  </div>
  <span
    className="item-description"
    title="Download link description"
  >
    Download link description
  </span>
</div>
`;
